xbmc/application/ApplicationActionListeners.h

```diff
@@ -15,12 +15,18 @@
 class CAction;
 class CApplication;
 class CCriticalSection;
+
+namespace KODI
+{
+namespace ACTION
```

Comment:
same here. I've seen a couple of "Actions" being introduced in different parts of the codebase, would it be beneficial to have this namespaced inside another INPUT namespace in order to make it implicit?

Reply:
My arguments against KEYBOARD in INPUT was:

* Names outside of INPUT get too long
* Keyboards are obviously input devices

The first applies here but not the second. However, the opposite is kinda true, Actions transcend different classifications. Indeed, for a class called ApplicationActionSomething, should it be in an INPUT namespace or an APPLICATION namespace? For this file in `xbmc/application/` it definitely should not be in the INPUT namespace.

After the refactor, only two headers directly in `xbmc/input/` use the generic INPUT namespace. Everything in subfolders uses a flat namespace of their own. This makes obvious sense for input devices and I see as fine-enough architecture. The only outlier is the `action/` subfolder, but I'd rather move actions out of input altogether than break this architectural pattern.

So I think the argument is, do we keep ACTION like this, or move the entire input code into subnamespaces under the INPUT namespace? I don't really see a gain for doing this currently, and I don't see much harm in not.

A general trend is moving classes deeper into a namespace structure (you'll rarely see the reverse). I think we can leave it as a future task to move classes further into namespaces.